### ISSUE TITLE
PATCH Replace operation on a complex multi-valued attribute is updating only the first matching record

### ISSUE THREAD

**Issue by jivpif** [simpleidserver/SimpleIdServer#65]
Hi,

While testing I found the following issue:

Given the data:

```
"roles": [
    {
        "primary": false,
        "type": "WindowsAzureActiveDirectoryRole",
        "value": "Member",
        "display": "Member"
    },
    {
        "primary": false,
        "type": "WindowsAzureActiveDirectoryRole",
        "value": "Admin",
        "display": "Admin"
    }
]
```

1. When executing any of the following Replace operations:
```
"op": "Replace",
"path": "roles.display",
"value": "New Display"
```
or
```
"op": "Replace",
"path": "roles[primary eq \"False\"].display",
"value": "New Display"
```

The result is:
```
"roles": [
    {
        "primary": false,
        "type": "WindowsAzureActiveDirectoryRole",
        "value": "Member",
        "display": "New Display"
    },
    {
        "primary": false,
        "type": "WindowsAzureActiveDirectoryRole",
        "value": "Admin"
    }
]
```

**The display attribute is updated only in the first matching record. It is removed from the rest of the matched roles.**
**Expected behaviour is the update to be applied in the same way on all matching roles records.**

2. When executing the following Replace operation:

```
"op": "Replace",
"path": "roles[primary eq \"False\"]",
"value": {
      "display": "New Display"
}
```

The result is:

```
roles": [
        {
            "primary": false,
            "type": "WindowsAzureActiveDirectoryRole",
            "value": "Admin",
            "display": "New Display"
        }
    ]
```

**The first matching record is updated. The rest of the matching records are removed.**
**Expected behaviour is the update to be applied in the same way on all matching roles records.**

Thanks in advance for the time

**Comment by simpleidserver** [comment]
The issue is fixed in the branch "release/1.1.9" and the UTS have been updated to check if all the sub attributes are updated.

**Comment by jivpif** [comment]
Thanks a lot for the quick fix again :) It works.
